lib/management.js

```diff
@@ -245,64 +290,72 @@ function saveServiceCredentials(conf, auth) {
     // initialized = true;
 }
 
-function loadOverlayVersion(db, version, cb) {
-    db.get(`configuration/overlay/${version}`, {}, (err, val) => {
+function loadOverlayVersion(metadata, version, cb) {
+    metadata.getObjectMD(serviceBucket, `configuration/overlay/${version}`, {},
+    logger, (err, val) => {
         if (err) {
             return cb(err);
         }
-        return cb(null, JSON.parse(val));
+        return cb(null, val);
     });
 }
 
 function initManagement(done) {
-    // if (initialized) {
-    //     return process.nextTick(done);
-    // }
-
-    const mdDb = mdClient.openDB(error => {
-        if (error) {
-            return process.nextTick(done, error);
-        }
-
-        const db = mdDb.openSub('PENSIEVE');
+    function iterate(done) {
+        return async.waterfall([
+            cb => {
+                if (!setup) {
+                    setup = true;
+                    metadata = new Metadata('mongodb', params, logger);
+                    return metadata.setup(() => cb());
+                }
+                return process.nextTick(cb());
+            },
+            cb => {
```

Comment:
Minor: for these steps in the waterfall we could save some lines by not using the braces (as done below https://github.com/scality/backbeat/pull/187/files#diff-c529f19cba932673dfc25817365dec76R334).